From f7cad0d4c46409a403f6fad5161e45742ec445b0 Mon Sep 17 00:00:00 2001
From: Apollon Oikonomopoulos <apoikos@debian.org>
Date: Mon, 14 Jan 2019 23:27:12 +0200
Subject: [PATCH] Puppet linter: use %{line} in the log format

`arc lint` on Puppet manifests is broken with modern versions of
puppet-lint because it uses an old log-format directive.

`%{linenumber}` had been deprecated since 2014[1] in favor of `%{line}`
and was eventually removed in puppet-lint 2.1.0[2].

Fix this by using `%{line}`, which should work for everyone these days.

[1] https://github.com/rodjek/puppet-lint/commit/dd77ecaab12dc516c45ca620212e0f45b6b5a628
[2] https://github.com/rodjek/puppet-lint/commit/870cc9577b1dd58c3453402880cbf9fd5900fc31
---
 src/lint/linter/ArcanistPuppetLintLinter.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lint/linter/ArcanistPuppetLintLinter.php b/src/lint/linter/ArcanistPuppetLintLinter.php
index 8336cd42..5b78ffd0 100644
--- a/src/lint/linter/ArcanistPuppetLintLinter.php
+++ b/src/lint/linter/ArcanistPuppetLintLinter.php
@@ -56,7 +56,7 @@ final class ArcanistPuppetLintLinter extends ArcanistExternalLinter {
     return array(
       '--error-level=all',
       sprintf('--log-format=%s', implode('|', array(
-        '%{linenumber}',
+        '%{line}',
         '%{column}',
         '%{kind}',
         '%{check}',
-- 
2.19.2

